File: agentguard/executors/mock_browser.py
from agentguard.executors.base import Executor
from agentguard.models import ActionRequest


class MockBrowserExecutor(Executor):
    def preview(self, action_request: ActionRequest) -> dict:
        return {"executor": "mock_browser", "summary": f"Would run {action_request.action_type} on {action_request.target_resource}"}

    def execute(self, action_request: ActionRequest, execution_token: str) -> dict:
        return {
            "executor": "mock_browser",
            "executed": True,
            "message": f"Mock executed {action_request.action_type} with parameters {action_request.parameters}",
        }

File: ts-agentguard/src/executors/mock-browser.ts
import { ActionRequest } from "../models";

function pythonRepr(value: unknown): string {
  if (value === null) {
    return "None";
  }
  if (typeof value === "boolean") {
    return value ? "True" : "False";
  }
  if (typeof value === "number") {
    return String(value);
  }
  if (typeof value === "string") {
    return `'${value}'`;
  }
  if (Array.isArray(value)) {
    return `[${value.map(pythonRepr).join(", ")}]`;
  }
  if (typeof value === "object" && value !== null) {
    return `{${Object.entries(value)
      .map(([key, entry]) => `${pythonRepr(key)}: ${pythonRepr(entry)}`)
      .join(", ")}}`;
  }
  return String(value);
}

export class MockBrowserExecutor {
  preview(actionRequest: ActionRequest): Record<string, unknown> {
    return {
      executor: "mock_browser",
      summary: `Would run ${actionRequest.action_type} on ${actionRequest.target_resource}`,
    };
  }

  execute(actionRequest: ActionRequest, executionToken: string): Record<string, unknown> {
    void executionToken;
    return {
      executor: "mock_browser",
      executed: true,
      message: `Mock executed ${actionRequest.action_type} with parameters ${pythonRepr(actionRequest.parameters)}`,
    };
  }
}
